valid.go

```diff
@@ -297,6 +300,9 @@ func (d *decoder) wellformedHead() (t cborType, ai byte, val uint64, err error)
 		}
 		val = uint64(binary.BigEndian.Uint16(d.data[d.off : d.off+2]))
 		d.off += 2
+		if err := d.acceptableFloat(float64(float16.Frombits(uint16(val)).Float32())); err != nil {
+			return 0, 0, 0, err
+		}
```

Comment:
We need to check CBOR type before converting `val` to float because `wellformedHead()` is used by all CBOR types.

```suggestion
	if t == cborTypePrimitives{
		if err := d.acceptableFloat(float64(float16.Frombits(uint16(val)).Float32())); err != nil {
			return 0, 0, 0, err
		}
	}
```